libyul/optimiser/StructuralSimplifier.cpp

```diff
@@ -133,8 +140,23 @@ void StructuralSimplifier::simplify(std::vector<yul::Statement>& _statements)
 
 				return s;
 			}
-			else
-				return {};
+
+			// Remove cases with empty body if no default case exists
+			auto const defaultCase = boost::find_if(
```

Comment:
Could we instead use something like
```
bool hasNonemptyDefault = boost::any(cases, [](Const const& _case) {
return (!_case.value) && !case.body.statements.empty();
});
```
?

Reply:
That might spare you the comment above ;-).

Reply:
I like your idea. I moved it into it's extra function anyway, but I'll use this